### PR TITLE
Add tests for type attribute values for rel=stylesheet links

### PR DESCRIPTION
**Pull request by rwlbuis**
This tests some edge cases.

### CODE REVIEW

**Review by annevk — APPROVED** [review]
This looks great, but makes me wonder what rules we actually use here. Are you interested in taking this further and also test parameters and such? E.g., is `text/css;invalid` a match or a non-match? It would be good to know if we need a string comparison or full blown MIME type parser a la https://mimesniff.spec.whatwg.org/#parsing-a-mime-type and say so in the HTML Standard. (It seems rather vague at the moment.)

File: html/semantics/document-metadata/the-link-element/link-type-attribute.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<link rel=match href=link-type-attribute-ref.html>
+<body>
```
Comment (annevk) [inline html/semantics/document-metadata/the-link-element/link-type-attribute.html]: Nit: maybe omit the body tags as you did in the ref. A browser that fails might think that has something to do with it, while it's meaningless.

**Comment by rwlbuis** [comment]
> This looks great, but makes me wonder what rules we actually use here. Are you interested in taking this further and also test parameters and such? E.g., is `text/css;invalid` a match or a non-match? It would be good to know if we need a string comparison or full blown MIME type parser a la https://mimesniff.spec.whatwg.org/#parsing-a-mime-type and say so in the HTML Standard. (It seems rather vague at the moment.)

Yes I will have a look at testing parameters. Thanks for the review!

**Comment by rwlbuis** [comment]
> This looks great, but makes me wonder what rules we actually use here. Are you interested in taking this further and also test parameters and such? E.g., is `text/css;invalid` a match or a non-match? It would be good to know if we need a string comparison or full blown MIME type parser a la https://mimesniff.spec.whatwg.org/#parsing-a-mime-type and say so in the HTML Standard. (It seems rather vague at the moment.)

Chromium does use a full blown MIME type parser, see:
https://cs.chromium.org/chromium/src/third_party/blink/renderer/core/html/link_style.cc?type=cs&q=StyleSheetTypeIsSupported&sq=package:chromium&g=0&l=32

From my tests of Firefox, it seems to do the same. This means the parsing is very tolerant, i.e. AFAIK everything after ; is basically discarded, so "text/css;" prefix is enough to make the type valid as a stylesheet. To me it makes sense to use the MIME type parser. Do you think you want to specify anything for this soon?

**Comment by annevk** [comment]
I'm not in a hurry, but okay with that outcome. I created https://github.com/whatwg/html/issues/5186 to track this.

**Comment by rwlbuis** [comment]
> I'm not in a hurry, but okay with that outcome. I created [whatwg/html#5186](https://github.com/whatwg/html/issues/5186) to track this.

Great, thanks. I think I want to update https://bugs.webkit.org/show_bug.cgi?id=88157 to use full blown MIME type parsing as well to match FF and Chrome, but I think the real tests can wait until [whatwg/html#5186] gets processed. So are you okay with landing this as-is?

**Pull request merged**